Unix/http/http.c

```diff
@@ -1702,11 +1702,20 @@ static MI_Result _CreateAddListenerSocket(
 
     /* Create listener socket */
     {
-        Addr_InitAny(&addr, port);
+        /* Try IPv6 at first */
+        Addr_InitAnyIPv6(&addr, port);
```

Comment:
shouldn't we be listening on both IPv4 and IPv6?
Devices can have both IPv4 and IPv6 connectivity giving preference to IPv6 will impact IPv4 connections.

Reply:
listen on IPv6  will also accept ipv4 and ipv6 connections, check this: https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_72/rzab6/xacceptboth.htm

Reply:
I mean `in6addr_any` will listen on both IPv6 and IPv4.
```
/********************************************************************/
      /* After the socket descriptor is created, a bind() function gets a */
      /* unique name for the socket.  In this example, the user sets the  */
      /* address to in6addr_any, which (by default) allows connections to */
      /* be established from any IPv4 or IPv6 client that specifies port  */
      /* 3005. (that is, the bind is done to both the IPv4 and IPv6 TCP/IP    */
      /* stacks).  This behavior can be modified using the IPPROTO_IPV6   */
      /* level socket option IPV6_V6ONLY if required.                      */
      /********************************************************************/
````

Reply:
I need to learn and do some investigation for `in6addr_any` http://man7.org/linux/man-pages/man7/ipv6.7.html
```
Linux 2.2 optionally implements the Internet Protocol, version 6.
       This man page contains a description of the IPv6 basic API as
       implemented by the Linux kernel and glibc 2.1.  The interface is
       based on the BSD sockets interface; see socket(7).

       The IPv6 API aims to be mostly compatible with the IPv4 API (see
       ip(7)).  Only differences are described in this man page.

       To bind an AF_INET6 socket to any process, the local address should
       be copied from the in6addr_any variable which has in6_addr type.  In
       static initializations, IN6ADDR_ANY_INIT may also be used, which
       expands to a constant expression.  Both of them are in network byte
       order.

       The IPv6 loopback address (::1) is available in the global
       in6addr_loopback variable.  For initializations,
       IN6ADDR_LOOPBACK_INIT should be used.

       IPv4 connections can be handled with the v6 API by using the
       v4-mapped-on-v6 address type; thus a program needs to support only
       this API type to support both protocols.  This is handled
       transparently by the address handling functions in the C library.

       IPv4 and IPv6 share the local port space.  When you get an IPv4
       connection or packet to an IPv6 socket, its source address will be
       mapped to v6 and it will be mapped to v6.
```

Reply:
can you check the behavior in some older OS like Fedora 24, Ubuntu 14 or 16 and see if listening to IPv6 only accepts connections for both IPv4 and IPv6?

Reply:
will check old OSs.